pkg/cluster/cert.go

```diff
@@ -228,6 +193,33 @@ func recreateSecret(ctx context.Context, c client.Client, existingSecret, newSec
 	return nil
 }
 
+// generateCertSecret creates a secret if it does not exist;  recreate this secret if type not match; update data if outdated.
+func generateCertSecret(ctx context.Context, c client.Client, certSecret *corev1.Secret, secretName, namespace string) error {
+	existingSecret := &corev1.Secret{}
+	err := c.Get(ctx, client.ObjectKey{Name: secretName, Namespace: namespace}, existingSecret)
+	if err != nil {
+		if k8serr.IsNotFound(err) {
+			// Secret does not exist, create it
+			if createErr := c.Create(ctx, certSecret); createErr != nil {
+				return fmt.Errorf("failed creating certificate secret: %w", createErr)
+			}
+		} else {
+			return fmt.Errorf("failed getting certificate secret: %w", err)
+		}
+	} else if existingSecret.Type != certSecret.Type {
+		// Secret exists but with a different type, delete and create it again
+		if recreateSecret(ctx, c, existingSecret, certSecret) != nil {
+			return errors.New("failed to recreate secret with type corrected")
+		}
+	}
+	if isSecretOutdated(existingSecret.Data, certSecret.Data) {
```

Comment:
Yeah, in the original code it was part of only one function, just kept the functionality as is, had a thought that it should be in both :) Thanks!